tests/glsl-intrinsic/shader-subgroup/glsl-test-utils.slang

```diff
@@ -0,0 +1,56 @@
+//why macros and not templated generics? -- macros are faster to iterate (development) on
```

Comment:
Please avoid using macros on unit tests. One of the most important aspect of the unit test is that it must be easy to read. The use of macro makes things much harder for the other people to read and maintain.

Reply:
does slang support some version of function pointer passing?

Reply:
There might be a way to use pointer but you don't need it here.
It will be unnecessary complexity.